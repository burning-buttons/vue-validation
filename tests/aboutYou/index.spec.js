import Component from "@/pages/aboutYou";
import { mountFactory, flush } from "../../../store";
import {
  ROUTE_HEALTH_PT_1,
  MSG_ERROR_FUTURE_DATE,
  STORE_SAVE_CHARACTERISTIC,
} from "@/constants";

const characteristics = {
  dob_month: 1,
  dob_year: 2020,
  sex: 0,
  height: 186.6,
  ethnicity: "caucasian",
  coutry: "Russia",
  zipcode: "123",
};
const state = { characteristics };

describe("About you page", () => {
  let wrapper;

  beforeEach(() => ({ wrapper } = mountFactory({ Component })));

  it("renders correct", async () => {
    wrapper.find(".about-you-page__submit").trigger("click");
    await flush();

    expect(wrapper).toMatchSnapshot();
  });

  it("date cann't be in the future", async () => {
    const state = {
      characteristics: { dob_month: 1, dob_year: 3020 },
    };
    const { wrapper } = mountFactory({ Component, state });
    wrapper.find(".about-you-page__submit").trigger("click");
    await flush();

    expect(
      wrapper.find(".about-you-page__birthday .error-message").text()
    ).toBe(MSG_ERROR_FUTURE_DATE);
  });

  it("correct fields shouldn't have errors", async () => {
    const { wrapper } = mountFactory({ Component, state });
    wrapper.find(".about-you-page__submit").trigger("click");
    await flush();
    await wrapper.vm.$nextTick();

    expect(wrapper.find(".error-message").exists()).toBeFalsy();
  });

  it("submitting with correct fields changes route", async () => {
    const { wrapper, actions } = mountFactory({ Component, state });
    wrapper.find(".about-you-page__submit").trigger("click");
    await flush();

    expect(actions[STORE_SAVE_CHARACTERISTIC]).toHaveBeenCalledWith(
      expect.any(Object),
      characteristics
    );
    expect(wrapper.vm.$router.push).toHaveBeenCalledWith(
      { name: ROUTE_HEALTH_PT_1 },
      null,
      expect.any(Function)
    );
  });
});
