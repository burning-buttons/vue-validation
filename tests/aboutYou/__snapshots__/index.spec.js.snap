// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About you page renders correct 1`] = `
<div class="about-you-page">
  <div class="header-quiz">
    <div class="header-quiz__container container">
      <div class="stub-link header-quiz__logo" to="[object Object]"></div>
      <h1 class="header-quiz__title">title</h1>
      <!---->
    </div>
    <div class="progress">
      <div class="progress__ready"></div>
    </div>
  </div>
  <div class="container page__container">
    <div class="card">
      <!---->
      <div class="card__container">
        <!----> <span><div class="birth-day card__section about-you-page__birthday"><h3>In what month and year were you born (month/year)?</h3> <div class="birth-day__list"><span class="birth-day__item"><div class="select error"><input type="text" class="select__input"> <div class="select__view"><span class="select__placeholder select__view-text">Month</span></div>
      <!---->
      <!---->
    </div></span> <span class="birth-day__item"><div class="select error"><input type="text" class="select__input"> <div class="select__view"><span class="select__placeholder select__view-text">Year</span>
  </div>
  <!---->
  <!---->
</div></span></div> <span><input type="text" class="birth-day__full-date"> <div class="error-message mt">Invalid date</div></span></div>
<div class="card__section about-you-page__sex"><span><h3>What is your sex?</h3> <div class="radio-list"><label class="radio-list__item"><input type="radio" class="radio with-gap" value="1"> <span>Male</span></label><label class="radio-list__item"><input type="radio" class="radio with-gap" value="0"> <span>Female</span></label></div>
<div class="error-message mt">Field sex is required</div>
</span></div>
<div class="card__section about-you-page__height">
  <h3>What is your height?</h3>
  <div class="user-height"><span class="user-height__item"><div class="input error"><input type="text" placeholder="ft" class="input__node"></div> <div class="error-message mt">Field feet is required</div></span> <span class="user-height__item"><div class="input"><input type="text" placeholder="in" class="input__node"></div> <!----></span> <button class="button button_text user-height__toggler">
      Switch to metric
    </button></div> <span><input type="text" class="user-height__common-height"> <!----></span>
</div>
<div class="card__section about-you-page__ethnicity">
  <h3>What is your ethnicity?</h3> <span><div class="select error"><input type="text" class="select__input"> <div class="select__view"><span class="select__placeholder select__view-text">Select value</span>
</div>
<!---->
<!---->
</div>
<div class="error-message mt">Field ethnicity is required</div></span></div>
<div class="user-living card__section about-you-page__living">
  <h3>Where are you currently living?</h3> <span><div class="input error"><input type="text" placeholder="Zip Code / Postal Code" class="input__node"></div> <div class="error-message mt">Field zip/post code is required</div></span>
  <div class="user-living__link"><a href="#">Outside the US?</a></div>
</div>
<!---->
<div class="page__card-footer"><button class="button button_secondary page__card-footer-button about-you-page__submit"><span>Next</span> <span class="icon icon__arrow icon__arrow_right"></span></button></div></span>
<!---->
</div>
</div>
<!---->
</div>
</div>
`;
